routing/pathfind.go

```diff
@@ -204,6 +207,19 @@ func newRoute(sourceVertex route.Vertex,
 			tlvPayload = supports(
 				finalHop.features, lnwire.TLVOnionPayloadOptional,
 			)
+
+			// If the receiver supports TLV onion payloads and we
+			// have a payment address from the invoice, we'll go
+			// ahead and include an MPP record to gain the
+			// additional protections offered by MPP.
+			supportsPayAddr := supports(
+				finalHop.features, lnwire.PaymentAddrOptional,
+			)
+			if supportsPayAddr && finalHop.paymentAddr != nil {
+				mpp = record.NewMPP(
+					finalHop.amt, *finalHop.paymentAddr,
+				)
+			}
```

Comment:
Yes, we may want to rethink what happens where. If path finding becomes payload size aware, it will also need to do something similar.